src/test/java/com/scalar/database/storage/cassandra4driver/ClusterManagerTest.java

```diff
@@ -0,0 +1,224 @@
+import com.scalar.database.exception.storage.ConnectionException;
+import com.scalar.database.exception.storage.StorageRuntimeException;
+import java.util.Optional;
+import java.util.Properties;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** */
+public class ClusterManagerTest {
+  private static final String VALID_ADDRESS = "127.0.0.1";
+  private static final String INVALID_PORT = Integer.toString(Integer.MAX_VALUE);
+  private static final String ANY_USERNAME = "username";
+  private static final String ANY_PASSWORD = "password";
+  private static final String ANY_KEYSPACE_NAME = "keyspace";
+  private static final String ANY_TABLE_NAME = "table";
+  @Mock private CqlSession session;
+  @Mock private CqlSessionBuilder builder;
+  private ClusterManager manager;
+  private Properties props;
+  private DatabaseConfig config;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    props = new Properties();
+    props.setProperty(DatabaseConfig.CONTACT_POINTS, VALID_ADDRESS);
+    props.setProperty(DatabaseConfig.USERNAME, ANY_USERNAME);
+    props.setProperty(DatabaseConfig.PASSWORD, ANY_PASSWORD);
+    config = new DatabaseConfig(props);
+
+    manager = Mockito.spy(new ClusterManager(config));
+    doReturn(builder).when(manager).getBuilder();
+    when(builder.build()).thenReturn(session);
+  }
+
+  @Test
+  public void getSession_CalledOnce_ShouldConnectAndReturn() {
+    // Act Assert
+    assertThatCode(
+            () -> {
+              manager.getSession();
+            })
+        .doesNotThrowAnyException();
+
+    // Assert
+    verify(manager).getBuilder();
+    verify(builder).build();
+  }
+
+  @Test
+  public void getSession_ConfigWithInvalidPortGiven_ShouldThrowExceptionProperly() {
+    // Arrange
+    props.setProperty(DatabaseConfig.CONTACT_PORT, INVALID_PORT);
+    config = new DatabaseConfig(props);
+    manager = Mockito.spy(new ClusterManager(config));
+
+    // Act Assert
+    assertThatThrownBy(
+            () -> {
+              manager.getSession();
+            })
+        .isInstanceOf(ConnectionException.class)
+        .hasCauseExactlyInstanceOf(IllegalArgumentException.class);
+
+    // Assert
+    verify(builder, never()).build();
+  }
+
+  @Test
+  public void getSession_CalledOnceAndDriverExceptionThrown_ShouldThrowConnectionException() {
+    // Arrange
+    DriverException toThrow = mock(DriverException.class);
+    when(builder.build()).thenThrow(toThrow);
+
+    // Act Assert
+    assertThatThrownBy(
+            () -> {
+              manager.getSession();
+            })
+        .isInstanceOf(ConnectionException.class)
+        .hasCause(toThrow);
+
+    // Assert
+    verify(manager).getBuilder();
+    verify(builder).build();
+  }
+
+  @Test
+  public void getSession_CalledAfterBuilt_ShouldNotBuild() {
+    // Arrange
+    manager.getSession();
+    manager.close();
```

Comment:
I think it is testing a little different thing from what it used to. 
so, maybe we should rename the test to something like `getSession_CalledAfterClosed_ShouldBuildAgain()`.

I feel it's a bit weird to see `getBuilder()` is called only once even after the cluster is closed. (it's related to the comment about `close()` method above.)
From users' perspective, I feel the expected behavior would be calling `getBuilder` and `build()` both twice ?
 
Whether or not we need to keep testing the original behavior, I feel we might not need to do it since there is no build() method in the class.

Reply:
I renamed the test name. I think this test can check `close()` (setting null).